src/hats/catalog/catalog_collection.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+from upath import UPath
+
+from hats.catalog import Catalog
+from hats.catalog.dataset.collection_properties import CollectionProperties
+from hats.pixel_math import HealpixPixel
+
+
+class CatalogCollection:
+    """A collection of HATS Catalog with data stored in a HEALPix Hive partitioned structure
+
+    Catalogs of this type are described by a `collection.properties` file which specifies
+    the underlying main catalog, margin catalog and index catalog paths. These catalogs are
+    stored at the root of the collection, each in its separate directory::
+
+        catalog_collection/
+        ├── main_catalog/
+        ├── margin_catalog/
+        ├── index_catalog/
+        ├── collection.properties
+
+    Margin and index catalogs are optional but there could also be multiple of them. The
+    catalogs loaded by default are specified in the `collection.properties` file in the
```

Comment:
```suggestion
    catalogs used by default are specified in the `collection.properties` file in the
```
Not necessarily "loaded"